R/benchmark.R

```diff
@@ -136,10 +136,33 @@ Benchmark <- function(name,
 #' @export
 BenchEnvironment <- function(...) list2env(list(...))
 
-default_params <- function(bm, ...) {
+#' Generate a dataframe of default parameters for a benchmark
+#'
+#' Generates a dataframe of parameter combinations for a benchmark to try based
+#' on the parameter defaults of its `setup` function and supplied parameters.
+#'
+#' @param x An object for which to generate parameters
+#' @param ... Named arguments corresponding to the parameters of `bm`'s `setup`
+#' function. May also contain `cpu_count`, `lib_path`, and `mem_alloc`.
+#'
+#' @return For `default_params.Benchmark`, a dataframe of parameter combinations
+#' to try with a column for each parameter and a row for each combination.
+#'
+#' @export
+default_params <- function(x, ...) {
+  UseMethod("default_params")
+}
```

Comment:
Since this will be exported, do we want to rename this to `get_default_parameters()` or something? We don't need to; it's just a convenient time to do so if we want

Reply:
Yes, IMO

Reply:
done, felt like time.